File: python_cpp/uPIMulator_frontend/src/abi/directive/byte_directive.py
from abi.word.immediate import Immediate
from abi.word.representation import Representation


class ByteDirective:
    def __init__(self, value: int):
        self._value: Immediate = Immediate(Representation.UNSIGNED, 8 * ByteDirective.size(), value)

    def value(self) -> int:
        return self._value.value()

    @staticmethod
    def size() -> int:
        return 1

File: golang/uPIMulator/src/linker/kernel/directive/byte_directive.go
package directive

import (
	"uPIMulator/src/abi/encoding"
	"uPIMulator/src/abi/word"
)

type ByteDirective struct {
	immediate *word.Immediate
}

func (this *ByteDirective) Init(value int64) {
	this.immediate = new(word.Immediate)
	this.immediate.Init(word.UNSIGNED, 8*int(this.Size()), value)
}

func (this *ByteDirective) Size() int64 {
	return 1
}

func (this *ByteDirective) Immediate() *word.Immediate {
	return this.immediate
}

func (this *ByteDirective) Encode() *encoding.ByteStream {
	return this.immediate.ToByteStream()
}
